doc/adr/0009-load-and-read-streams-in-same-transaction.md

```diff
@@ -0,0 +1,44 @@
+# 9. Load and Read Streams in the Same Transaction
+
+Date: 20/01/2017
+
+## Status
+
+Proposed
+
+## Context
+
+Reading streams from within the same transaction that calls loadStream() must
+be supported. There are a number of benefits to doing this rather than reading
+streams outside the transaction.
+
+1. If the transaction that calls loadStream() is run with retry, you risk
+   leaking resources because multiple streams may be created that never get
+   returned to the caller (so the caller has no opportunity to close them).
+   The stream can not only be read (entirely) inside the transaction, but
+   also closed inside the transaction. This makes it safe to run with retries
+   (since the retries can also close the stream they load).
+
+2. You risk having a transaction succeed (commit) where the stream is loaded,
+   but at some time later when the actual steam data is read that read can
```

Comment:
also, "is read that read" scans badly - maybe "when the actual stream data is read, that read can fail"